cumulus/client/consensus/aura/src/collators/basic.rs

```diff
@@ -215,11 +216,13 @@ where
 			//
 			// Most parachains currently run with 12 seconds slots and thus, they would try to
 			// produce multiple blocks per slot which very likely would fail on chain. Thus, we have
-			// this "hack" to only produce on block per slot.
+			// this "hack" to only produce on block per slot per relay chain fork.
 			//
 			// With https://github.com/paritytech/polkadot-sdk/issues/3168 this implementation will be
 			// obsolete and also the underlying issue will be fixed.
-			if last_processed_slot >= *claim.slot() {
+			if last_processed_slot >= *claim.slot() &&
+				last_relay_chain_block > *relay_parent_header.number()
```

Comment:
Maybe my brain is playing me, but I think it needs to be inverted.
```suggestion
				last_relay_chain_block < *relay_parent_header.number()
```

Basically we want to skip if we are in the same slot as the previously build block and the relay parent we are building on has a higher number than the last one. If we are in the same slot but the relay parent has the same number as the last one, we are in a fork and want to build.